permission/permission.go

```diff
@@ -0,0 +1,759 @@
+package permission
+
+import (
+	"crypto/ecdsa"
+	"encoding/json"
+	"errors"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/eth"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethereum/go-ethereum/params"
+	pbind "github.com/ethereum/go-ethereum/permission/bind"
+	"github.com/ethereum/go-ethereum/raft"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+type NodeOperation uint8
+
+const (
+	NodeAdd NodeOperation = iota
+	NodeDelete
+)
+
+// permission config for bootstrapping
+type PermissionLocalConfig struct {
+	UpgrdAddress   string `json:"upgrdableAddress"`
+	InterfAddress  string `json:"interfaceAddress"`
+	ImplAddress    string `json:"implAddress"`
+	NodeAddress    string `json:"nodeMgrAddress"`
+	AccountAddress string `json:"accountMgrAddress"`
+	RoleAddress    string `json:"roleMgrAddress"`
+	VoterAddress   string `json:"voterMgrAddress"`
+	OrgAddress     string `json:"orgMgrAddress"`
+	NwAdminOrg     string `json:"nwAdminOrg"`
+	NwAdminRole    string `json:"nwAdminRole"`
+	OrgAdminRole   string `json:"orgAdminRole"`
+
+	Accounts      []string `json:"accounts"` //initial list of account that need full access
+	SubOrgBreadth string   `json:"subOrgBreadth"`
+	SubOrgDepth   string   `json:"subOrgDepth"`
+}
+
+type PermissionCtrl struct {
+	node             *node.Node
+	ethClnt          *ethclient.Client
+	eth              *eth.Ethereum
+	isRaft           bool
```

Comment:
Is there a better way to remove this flag? Ideally we want to be consensus agnostic

Reply:
This flag is needed as whenever the node is deactivated or blacklisted, we have to disconnect the node from current network. For Raft we need to disconnect from Raft cluster and P2P.